src/Build/Globbing/MSBuildGlob.cs

```diff
@@ -117,23 +117,20 @@ public MatchInfoResult MatchInfo(string stringToMatch)
         {
             ErrorUtilities.VerifyThrowArgumentNull(stringToMatch, nameof(stringToMatch));
 
-            if (FileUtilities.PathIsInvalid(stringToMatch) ||
-                !IsLegal)
+            if (FileUtilities.PathIsInvalid(stringToMatch) || !IsLegal)
             {
                 return MatchInfoResult.Empty;
             }
 
             var normalizedInput = NormalizeMatchInput(stringToMatch);
 
-            bool isMatch;
-            string fixedDirectoryPart, wildcardDirectoryPart, filenamePart;
             FileMatcher.GetRegexMatchInfo(
                 normalizedInput,
                 _state.Value.Regex,
-                out isMatch,
-                out fixedDirectoryPart,
-                out wildcardDirectoryPart,
-                out filenamePart);
+                out bool isMatch,
```

Comment:
minor: is 4 out parameters better than a new encapsulation of the same state?

Reply:
Unfortunately cannot do what you're suggesting as `GetRegexMatchInfo` lives in src/shared, `MSBuildGlob` in src/Build, and `MatchInfoResult` is a public type so it can't be easily moved. I'd say 4 output parameters, while a minor readability hit, is borderline acceptable. There are two calls to `GetRegexMatchInfo`, both populating their own structure with the returned data. Adding another representation just for the purpose of returning it from the method would IMO be an overkill.